crates/turborepo-auth/src/auth/logout.rs

```diff
@@ -1,5 +1,34 @@
-use turborepo_ui::{cprintln, GREY, UI};
+use std::fs;
 
-pub fn logout(ui: &UI) {
-    cprintln!(ui, GREY, ">>> Logged out");
+use tracing::error;
+use turborepo_api_client::Client;
+use turborepo_ui::{cprintln, GREY};
+
+use crate::{Error, LogoutOptions};
+
+pub fn logout<T: Client>(options: &LogoutOptions<T>) -> Result<(), Error> {
+    if let Err(err) = remove_token(options) {
+        error!("could not logout. Something went wrong: {}", err);
+        return Err(err);
+    }
+
+    cprintln!(options.ui, GREY, ">>> Logged out");
+    Ok(())
+}
+
+fn remove_token<T: Client>(options: &LogoutOptions<T>) -> Result<(), Error> {
+    let global_config_path = options.path;
+    let content =
+        fs::read_to_string(global_config_path).map_err(|e| Error::FailedToReadAuthFilePath {
+            path: global_config_path.clone(),
+            error: e,
+        })?;
+
+    let mut data: serde_json::Value = serde_json::from_str(&content)?;
+    data["token"] = serde_json::Value::Null;
+
+    let new_content = serde_json::to_string_pretty(&data)?;
+    fs::write(global_config_path, new_content)?;
```

Comment:
If we don't care about ordering and stick with using `serde` we should again use the turbopath variant
```suggestion
    global_config_path.create_with_contents(new_content)?;
```